VRDR/IJEMortality.cs

```diff
@@ -2491,21 +2492,57 @@ public string IDOB_YR
             }
         }
 
-        /// <summary>Infant Death/Birth Linking - year of birth</summary>
+        /// <summary>Infant Death/Birth Linking - Birth state</summary>
         [IJEField(90, 671, 2, "Infant Death/Birth Linking - State, U.S. Territory or Canadian Province of Birth - code", "BSTATE", 1)]
         public string BSTATE
         {
             get
             {
                 if (!String.IsNullOrWhiteSpace(BCNO))
                 {
-                    return Dictionary_Geo_Get("BSTATE", "PlaceOfBirth", "address", "state", true);
+                    String state = Dictionary_Get_Full("BSTATE", "BirthRecordState", "code");
+                    String retState;
+                // If the country is US or CA, strip the prefix
+                    if (state.StartsWith("US-") || state.StartsWith("CA-"))
+                    {
+                        // Console.Error.WriteLine("Starts with US- state= " + state + "retState =" + state.Substring(2));
+                        retState = state.Substring(3);
+                    } else {
+                        // Console.Error.WriteLine("No Prefix state= " + state);
+                        retState = state;
+                    }
+                    // Console.Error.WriteLine("BSTATE.get returning " + retState + ",  state = " + state);
+                    return retState;
                 }
                 return ""; // Blank
             }
             set
             {
-                // NOOP
+                if (!String.IsNullOrWhiteSpace(value))
+                {
+                    String birthCountry = BPLACE_CNT;
+                    String ISO31662code;
+                    Console.Error.WriteLine("BSTATE.set with country = " + birthCountry);
+                    switch (value){
+                        case "ZZ": // UNKNOWN OR BLANK U.S. STATE OR TERRITORY OR UNKNOWN CANADIAN PROVINCE OR UNKNOWN/ UNCLASSIFIABLE COUNTRY
+                            // Console.Error.WriteLine("BSTATE.set with value ZZ");
+                            return;  // do nothing 
```

Comment:
If this setter is called on a record which already has a state set will it clear the state or leave it set?